### DISCUSSION TITLE
Embed AtlasMap into webapplication

### DISCUSSION THREAD

**Discussion by assimbly** [atlasmap/atlasmap#4057] [Q&A]
I understand that AtlasMap standalone has a Spring Boot backend and React frontend. Now I would like to only use the UI part and embed the UI (mapper) into my own webapplications (I have both Angular and React frontends). On these webapps users can use AtlasMap and then I would like to save the ADM file to a database.

On npmjs.com I saw that there are two packages:

https://www.npmjs.com/package/@atlasmap/core
https://www.npmjs.com/package/@atlasmap/atlasmap

There are also two other packages, but they seem outdated.

I saw some earlier discussion on this topic like

1. [How can I use this package for angular.](https://github.com/atlasmap/atlasmap/discussions/3383)
2. [Generating Atlasmap component as a web component](https://github.com/atlasmap/atlasmap/discussions/3614)

As I have the same use case, there is obvious demand for this (especially on the Angular side). It would be nice to embed AtlasMap just like for example CodeMirror can be embedded into a webapp.

Now my questions are:

1. Is there any documentation, tutorials or (Github) examples on this topic?

I don't want to reinvent the wheel, but I could find anything in the developer guide. I searched for example for "embedding" and similar words in the [developer guide](https://docs.atlasmap.io/developer-guide), but couldn't find anything. 

2. Is the [standalone](https://github.com/atlasmap/atlasmap/tree/main/ui/packages/atlasmap-standalone) version somewhat of a reference implementation?

3. Which of the NPM packages are needed to embed the UI into a webapplication? Only core or both?

**Comment by igarashitm** [comment]
I don't think there is such, but the real example is the Syndesis. Syndesis embeds AtlasMap UI into its own UI.
https://github.com/syndesisio/syndesis/tree/1.14.x/app/ui-react/packages/atlasmap-adapter

In theory `@atlasmap/core` is the UI side business logic, and `@atlasmap/atlasmap` is a set of React components. Not a few code doesn't follow the theory ATM, and we've been refactoring them.

**Reply by skin27** [reply]
Thanks, the adapters are indeed a good reference. And good to know refactoring is on going.

In case you want to test embedding AtlasMap in several frontends you could try [JHipster](https://www.jhipster.tech/). Then you can set up generic React, Angular, Vue apps in minutes.
